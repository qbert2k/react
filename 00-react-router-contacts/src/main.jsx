import React from "react";
import ReactDOM from "react-dom/client";
import {createBrowserRouter, RouterProvider} from "react-router-dom";
import "./index.css";
import ErrorPage from "./error-page.jsx";
import Root, {
    loader as rootLoader,
    action as rootAction,
} from "./routes/root.jsx";
import Contact, {
    loader as contactLoader,
} from "./routes/contact.jsx";

const router = createBrowserRouter([
    {
        path: "/",
        element: <Root/>,
        errorElement: <ErrorPage/>,
        loader: rootLoader,
        action: rootAction,
        children: [
            {
                path: "contacts/:contactId",
                element: <Contact/>,
                loader: contactLoader,
            }
        ]
    }
]);

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <RouterProvider router={router}/>
    </React.StrictMode>
);