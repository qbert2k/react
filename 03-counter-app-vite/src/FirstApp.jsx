const newMessage = {
    message: 'Hola Mundo',
    title: 'title'
};

const getResult = (a, b) => a + b;

export const FirstApp = () => {
    return (<>
        <h1>{newMessage.title}</h1>
        <code>{JSON.stringify(newMessage)}</code>
        <p>Soy un texto</p>

        <p>{getResult(5, 6)}</p>
    </>);
}