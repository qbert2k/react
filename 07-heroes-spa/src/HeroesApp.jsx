import {AppRouter} from './router/AppRouter.jsx';

export const HeroesApp = () => {
    return (
        <>
            <AppRouter/>
        </>
    );
}