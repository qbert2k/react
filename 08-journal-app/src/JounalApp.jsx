export const JounalApp = () => {
    return (
        <h1>Journal App</h1>
    );
}