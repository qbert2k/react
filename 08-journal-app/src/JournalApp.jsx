import {AppRouter} from './router/AppRouter.jsx';
import {AppTheme} from './theme/index.js';

export const JournalApp = () => {
    return (
        <AppTheme>
            <AppRouter/>
        </AppTheme>
    );
}