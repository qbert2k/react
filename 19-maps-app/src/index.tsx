import React from "react";
import ReactDOM from "react-dom/client";
import {MapsApp} from "./MapsApp";
import mapboxgl from "mapbox-gl";

mapboxgl.accessToken = 'pk.eyJ1IjoicWJlcnQyayIsImEiOiJjajU2d2JpOGMwdGk4MnFyMHYyb203YmJ0In0.NFxFL1P8GCDSCmBP1oZQlA';

if (!navigator.geolocation) {
    alert('The browser has not Geolocation features');
    throw new Error('The browser has not Geolocation features');
}

const root = ReactDOM.createRoot(
    document.getElementById('root') as HTMLElement
);

root.render(
    <React.StrictMode>
        <MapsApp/>
    </React.StrictMode>
);