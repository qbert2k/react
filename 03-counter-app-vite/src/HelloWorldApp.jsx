import {FirstApp} from "./FirstApp.jsx";

export const HelloWorldApp = () => {
    return (
        <h1>Hola Mundo!!!</h1>
    );
}

// function App() {
//     return <h1>Hola Mundo!!!</h1>;
// }
//
// export default App;