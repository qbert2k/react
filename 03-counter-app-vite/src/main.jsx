import React from 'react';
import ReactDOM from 'react-dom/client';
import {FirstApp} from './FirstApp.jsx';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp/>
    </React.StrictMode>
)