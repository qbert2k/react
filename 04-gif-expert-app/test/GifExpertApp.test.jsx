import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Test component <GifExpertApp />', () => {

    test('should match snapshot', () => {
        const {container} = render( <GifExpertApp /> );
        expect(container).toMatchSnapshot();
    });
});