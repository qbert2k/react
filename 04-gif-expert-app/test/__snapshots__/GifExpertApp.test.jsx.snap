// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifExpertApp /> should match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
